src/benchmark_runner.cc

```diff
@@ -445,28 +485,15 @@ void BenchmarkRunner::DoOneRepetition() {
            "then we should have accepted the current iteration run.");
   }
 
-  // Oh, one last thing, we need to also produce the 'memory measurements'..
+  // Produce memory measurements if requested.
   MemoryManager::Result* memory_result = nullptr;
   IterationCount memory_iterations = 0;
```

Comment:
this breaks reporting.  you need to _return_ the number of memory iterations for l501 to make sense.  ie, outside the call to RunMemoryManager you should be setting the memory_iterations to min(16, iters) so that the reporting lower down can report the right thing.  or make IterationCount a pointer/reference but don't do that :)

Reply:
Done